libraries/AdaptiveExpressions/BuiltinFunctions/GreaterThan.cs

```diff
@@ -18,14 +19,33 @@ public GreaterThan()
             : base(
                   ExpressionType.GreaterThan,
                   Function,
-                  FunctionUtils.ValidateBinaryNumberOrString,
-                  FunctionUtils.VerifyNumberOrString)
+                  FunctionUtils.ValidateBinary,
+                  FunctionUtils.VerifyNumberOrStringOrDateTime)
         {
         }
 
         private static bool Function(IReadOnlyList<object> args)
         {
-            return FunctionUtils.CultureInvariantDoubleConvert(args[0]) > FunctionUtils.CultureInvariantDoubleConvert(args[1]);
+            if (args[0].IsNumber())
+            {
+                return FunctionUtils.CultureInvariantDoubleConvert(args[0]) > FunctionUtils.CultureInvariantDoubleConvert(args[1]);
```

Comment:
number is also IComparable

Reply:
CompareTo method requires two object have the same type strictly. 
1.CompareTo(1.2) will throw an error.
So, I compare numbers first. Otherwise fallback to IComparable.